### ISSUE TITLE
New integration

### ISSUE THREAD

**Issue by AmandaBekhoj** [mamont77/skillspeople#148]
there is going to be some changes going forward, 

one of them is, that we are implementing a new recruitment system called Zoho, and we need to integrate it with drupal :)

i just have one task for now, but Christian will explain what should be done later :)

**so, task one:**

you remember how i wanted to a new receiver field for all the job pages? and also, that a 'hiring manager' should also get the application of someone applied for the job? (#145)

we now need the recipient of the application to be hnsrdv@resumes.zohorecruit.com 
this email is connected to our new CRM system Zoho, where it will be possible for us to search in the applications and CVs :) 

also, there need to be a difference in the subject line. 
for the email that is going out to the hiring manager, the subject line should read: Ny candidate far SkillsPeople til stillingen [insert short job title]' 

for the mail that is going to hnsrdv@resumes.zohorecruit.com, the subject should have the Zoho ID plus source (source should be SkillsPeople.dk)
i think you can find correct information about how to integrate here https://www.zoho.com/recruit/help/resume-inbox-format.html 

i will be updating the Zoho IDs on the job - but for now it will only be digicure and coresec :)

i know this was a lot of messy information, i hope it makes sense :)

**Comment by mamont77** [comment]
>>i will be updating the Zoho IDs on the job - but for now it will only be digicure and coresec :)

Will it instead Podio ID?
because "henvis_kandidat" (http://www.skillspeople.dk/henvis-potentiel-kandidat) uses Podio ID for sending data to podio and will broken ))

Or need a new " Zoho ID" field?

**Comment by AmandaBekhoj** [comment]
we need the information be sent to Zoho from now on, we do not want to use podio anymore (i know, it is going to be very troublesome)

**Comment by mamont77** [comment]
So i can rename podio id to job id and delete a part of old podio code. Right?

**Comment by AmandaBekhoj** [comment]
yes, just call it job id - i already changed the id for coresec and digicure :)

**Comment by mamont77** [comment]
Can we delete this page? http://www.skillspeople.dk/henvis-potentiel-kandidat Or make modifications.

**Comment by AmandaBekhoj** [comment]
yea, we need to make modifications to it, since it disconnected to podio. 
:) so it should be the Zoho id instead..
the advisor id is still the same thing :)

**Comment by mamont77** [comment]
What do you think - there are other things that we could forget?

**Comment by mamont77** [comment]
I made some changes for "Søg stillingen". As described in the header of this task.

But I do not know how to check this. If I fill the form - real email sent to real people + zoho.

briefly logic:
1. user (anonymous or advisor) fill the form (+attached resume - optional) and push "send" btn.
2. drupap send mails

recipient
```php
$to = 'hnsrdv@resumes.zohorecruit.com';
```
text of mail
```php
  // text of mail
  $body[] =
    'Name: ' . $valid_advisor_first_name . ' ' . $valid_advisor_last_name . "<br />\r\n" .
    'Email: ' . $valid_advisor_drupal_mail . "<br />\r\n" .
    'Advisor Podio ID: ' . $valid_advisor_podio_id . "<br />\r\n" .
    'Job ID: ' . $valid_job_podio_id . "<br />\r\n" .
    'Job title: ' . $valid_job_title . "<br />\r\n" .
    'Request URL: http://www.skillspeople.dk' . request_uri();
  $params = array(
    'body' => $body,
    'subject' => 'JobId: ' . $valid_job_podio_id . ', SkillsPeople.dk',
  );
```

then

recipient
```php
  $to = '';
  if ($talent_manager_mail) {
    $to .= ', ' . $talent_manager_mail;
  }
  if ($manager_mail) {
    $to .= ', ' . $manager_mail;
  }
```
body
```php
  $params = array(
    'body' => $body,
    'subject' => 'Ny candidate far SkillsPeople til stillingen ' . $valid_job_title,
  );
```

and then

recipient
```php
  $to = $valid_advisor_drupal_mail;
```
body
```php
  $params = array(
    'body' => $body,
    'subject' => 'Job: ' . $valid_job_title . ', SkillsPeople.dk',
  );
```

I know this is confusing to you. you can just check the values of subject and body. 

Total sent 3-4 mails at a time.

**Comment by mamont77** [comment]
>>i will be updating the Zoho IDs on the job - but for now it will only be digicure and coresec :)

i can see job id = 3 (http://www.skillspeople.dk/node/136/edit).This is not a mistake?

**Comment by mamont77** [comment]
Amanda, we're doing wrong. I am afraid that we will break the site.

Usually such change is made step by step.
Create new field (job id), some time working both fields (podio id and job id), other functional changes associated with the new field.  but old functionality still working. and only after some time removed the old functionality and field.

Currently this page (http://www.skillspeople.dk/henvis-potentiel-kandidat) contains mix of values (Zoho ID or Podio ID). If the user submits this form - Dupal does not know where to send data. Drupal can not know about any ID, it just value for Drupal.

I think we  need to create a "milestone" called "Zoho Integration". Create a complete list of tasks with a detailed description. And only then do something. In this case, the website would work without errors.

**Comment by AmandaBekhoj** [comment]
yea i figured this would be difficult to change, since we need to make all the integrations at the same time! 

i know there are different ID's on the site right now, but it is because there are some jobs, that we are finalizing, and they should not be integrated with zoho. 

yes, the new ID's are short, 2 and 3 are correct IDs. 

so you think i should change the ID's back to Podio IDs, and then we make a concrete list of tasks that need to be done?

**Comment by mamont77** [comment]
If it does not break the existing forms/blocks, you can not change anything back.

**Comment by AmandaBekhoj** [comment]
i do not understand. 

we need information from druptal to get over to Zoho, but only for http://www.skillspeople.dk/job/senior-sales-specialist-silkeborg-coresec-systems and http://www.skillspeople.dk/job/key-account-manager-lyngby-digicure, and for future jobs - as described in the issue description. 

If it makes things easier, we can unpublished the http://www.skillspeople.dk/henvis-potentiel-kandidat site when making changes?

Henvis Kandidat! (left block) - should go to Zoho
Henvis Kandidat! (separate page) - momentarily unpublished
Søg stillingen - should go to Zoho
Del med dit netværk - has this somethign to do with Zoho?

**Comment by mamont77** [comment]
Amanda, I just want to tell you that Drupal does not see the difference between 3 (Zolo ID) and 8970DB1E8E (Podio ID). It's just a string or a number.

Today I made changes for "Søg stillingen - should go to Zoho".
Imagine for a moment that the user has come to this page http://www.skillspeople.dk/job/sharepoint-account-manager-koebenhavn-innofactor and submitted "Søg stillingen" with 8970DB1E8E.

An mail will be sent to 'hnsrdv@resumes.zohorecruit.com' too. what happens at zoho in this case?

Now more clearly what I mean? :-)

>>Henvis Kandidat! (separate page) - momentarily unpublished

unpublished - done.

**Comment by mamont77** [comment]
Amanda, 
I filed a new issue for "Henvis Kandidat! (left block)". Please review. 
https://github.com/mamont77/skillspeople/issues/149
A little later I will describe the logic and functional and changes for the other two blocks.

**Comment by AmandaBekhoj** [comment]
1. when the ID is 8970DB1E8E, then it does not go to Zoho .. (but i will remove the jobs, that does not have a Zoho ID within this week)

2. we want to make podio obsolete, so the process of getting information to Zoho instead of podio can begin - on all parameters. What information about Zoho do you need to make this happen, do you need access?

**Comment by mamont77** [comment]
Amanda, please review and confirm #150 and #151.

**Comment by mamont77** [comment]
>>  What information about Zoho do you need to make this happen, do you need access?

If it possible i want to test my changes on drupal. To send 2-10 test mails to Zoho, and delete later. Please any access to Zoho.

**Comment by AmandaBekhoj** [comment]
login is: cj@skillspeople.dk
password: SuperMario

**Comment by mamont77** [comment]
Amanda,
I close this issue. We have created to individual tasks.
